q2_sample_classifier/classify.py

```diff
@@ -37,6 +37,70 @@
 }
 
 
+def metatable(ctx,
+              metadata,
+              table=None,
+              missing_samples='ignore',
+              missing_values='error'):
+    # gather numeric metadata
+    metadata = metadata.filter_columns(
+        column_type='numeric', drop_all_unique=True, drop_zero_variance=True,
+        drop_all_missing=True).to_dataframe()
+    # drop columns with negative values
+    metadata = metadata[-(metadata < 0).any(axis=1)]
+
+    if missing_values == 'drop_samples':
+        metadata = metadata.dropna(axis=0)
+    elif missing_values == 'drop_features':
+        metadata = metadata.dropna(axis=1)
+    elif missing_values == 'error' and metadata.isnull().values.any():
+        raise ValueError('You are attempting to coerce metadata containing '
+                         'missing values into a feature table! These may '
+                         'cause fatal errors downstream and must be removed '
+                         'or converted to 0. See the missing_values parameter '
+                         'to review your options.')
+    elif missing_values == 'fill':
+        metadata = metadata.fillna(0.)
+
+    if len(metadata.columns) == 0:
+        raise ValueError('All metadata columns have been filtered.')
+    if len(metadata.index) == 0:
+        raise ValueError('All metadata samples have been filtered.')
+    print(metadata)
```

Comment:
Are these print lines necessary, or just left-overs?

Reply:
oops — leftover from testing. thanks for catching @thermokarst ! removed in my most recent changes